### PR TITLE
fix: make `.extract_manifest()` also compile if needed

### PR DESCRIPTION
**Pull request by antazoey**
### What I did

pre 0.8. this happened, and I don't see why it should not happen, so making it happen again.
(here you go @Ninjagod1251 )

### Checklist

- [ ] All changes are completed
- [ ] New test cases have been added
- [ ] Documentation has been updated

### CODE REVIEW

**Comment by antazoey** [comment]
Some manual testing showing it working as well:

```python
In [7]: project.manifest.contract_types = {}

In [8]: mani_pedi = project.extract_manifest()
INFO (ape-vyper): Compiling using Vyper compiler '0.4.0'.
Input:
        contracts/Token.vy

In [9]: mani_pedi = project.extract_manifest()

In [10]: mani_pedi = project.extract_manifest()
```

**Comment by antazoey** [comment]
- [ ] Make also happen in manifest projects

File: src/ape/managers/project.py [hunk 1]
```diff
@@ -1891,6 +1891,12 @@ def reconfigure(self, **overrides):
         self.account_manager.test_accounts.reset()
 
     def extract_manifest(self) -> PackageManifest:
+        # Ensure it compiled (at least attempt).
+        try:
+            self.load_contracts()
+        except CompilerError as err:
+            logger.debug(err)
```
Comment (fubuloubu) [inline src/ape/managers/project.py:1898]: Should this be higher? Maybe warning

**Review by fubuloubu — DISMISSED** [review]

File: src/ape/managers/project.py (hunk 1 above)
Comment (antazoey) [reply src/ape/managers/project.py:1898]: perhaps! I am just worried about dependencies trying to extract manifests from that aren't meant to be compiled.

Comment (antazoey) [reply src/ape/managers/project.py:1898]: i suppose they could just `.manifest` in that case.

Comment (antazoey) [reply src/ape/managers/project.py]: I changed it to warning but added a comment. let me know if that makes sense, along with my concerns.

**Review by Ninjagod1251 — APPROVED** [review]

File: src/ape/managers/project.py [hunk 2]
```diff
@@ -2461,7 +2461,7 @@ def extract_manifest(self) -> PackageManifest:
         sources = dict(self.sources)
         contract_types = {
             n: ct
-            for n, ct in (self.manifest.contract_types or {}).items()
+            for n, ct in self.load_contracts().items()
```
[the remark this answers is not in the document]
Comment (antazoey) [reply src/ape/managers/project.py]: you can delete the `.load_contracts()` line, yes (once 0.8.14 is out).
However, it also doesn't hurt to have it.

File: src/ape/managers/project.py (hunk 1 above)
Comment (fubuloubu) [reply src/ape/managers/project.py]: > perhaps! I am just worried about dependencies trying to extract manifests from that aren't meant to be compiled.

Wonder if it makes more sense as like `.enrich_manifest` or `.fully_compiled_manifest`, or something

**Pull request merged**